packages/react-core/src/utils/createContextUtilities.tsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+
+import { isUndefined, isString } from '@aws-amplify/ui';
+
+export const INVALID_OPTIONS_MESSAGE =
+  'an `errorMessage` or a `defaultValue` must be provided in `options`';
+
+type OptionsWithDefaultValue<ContextType extends {}> = {
+  contextName: string;
+  defaultValue: ContextType;
+};
+
+type OptionsWithErrorMessage = {
+  contextName: string;
+  errorMessage: string;
+};
```

Comment:
Could, but if a `defaultValue` is provided the `errorMessage` will never be used since the value of [context here](https://github.com/aws-amplify/amplify-ui/pull/4331/files#diff-cb0ee7822afb24d2ebff2497c2f2ab0b5ebf1f60e45b6ced31ddfcbfa41d6174R102) will never evaluate to `undefined`